accepted/targeting-packs-and-runtime-packs.md

```diff
@@ -0,0 +1,286 @@
+        Microsoft.ASPNetCore.App.Ref/3.0.0/...
+        Microsoft.WindowsDesktop.App.Ref/3.0.0/...
+        NETStandard.Library.Ref/2.1.0/...
+    
+    sdk/      <--  build toolsets
+        3.0.100/
+```
+
+Note that there are no .nuspecs,.nupkgs,.sha512,.metadata,.p7s files in the packs/ tree as it does not serve as a NuGet fallback folder or source. When assets are consumed from this global location, NuGet is not involved in any way. Note that that as NuGet requirements change over time (e.g. package signing), the contents of a fallback folder or source need to change to meet them, and we do not want to have to service older offline packs for compatibility with newer SDKs.
+
+Furthermore, just like `sdk/*` and `shared/*`, `packs/*` content is either included in the sdk .zip or installed via a native installer with appropriate ref-counting. This is in stark, deliberate contrast to the `sdk/NuGetFallbackFolder` that gets content unzipped from each SDKs .lzma, and that only grows over time no matter what you uninstall. The .lzma will be eliminated entirely and replaced by these packs.
+
+
+# Packs Delivered via NuGet
+
+The global scheme above allows for offline builds in common cases. For example, by bundling .NET Core 3.0 targeting packs with .NET Core 3.0.* SDKs, you can install the 3.0.* SDK and build framework-dependent 3.0 projects without pulling anything from the network. However, if you have projects that require a targeting pack that is not globally installed, then the SDK will instruct NuGet to download during restore and use it from the package cache instead of the global location. For this, a new NuGet package type is introduced that:
+
+* cannot be installed into a project via a standard package reference
+* cannot depend on other packages
+* cannot be depended upon by other packages
+* is not subject to standard nuget resolution rules
+
+Furthermore, when NuGet is used, the assets acquired are not cataloged in project.assets.json as normal assets that are subject to NuGet resolution rules. Insetead, the build will use the same logic as the global/offline case and simply redirect it to a downloaded folder in the NuGet package cache. 
+
+This will be implemented using a new "download only package" feature from NuGet: https://github.com/NuGet/Home/issues/7339
+
+
+## Package names
+
+These packages will not appear in the NuGet gallery, NuGet manager, project.assets.json, runtimeconfig.json, or basically anywhere user-facing. As such, the names are essentially implementation details. The naming requirements are:
+
+1. Must not overlap with traditional framework NuGet package names
+2. Given a shared framework name, one can derive the targeting pack package name
+3. Given a shared framework name + RID, one can derive the runtime pack and apphost pack name
+
+**Proposal**
+ * Targeting pack packages: \<shared framework name\>.Ref
+ * Runtime pack packages: \<shared framework name\>.Runtime.\<rid\>
```

Comment:
What if we decided to distribute the contents of $dotnet_root/shared/ via packs as well? We made a decision earlier that shared frameworks and self-contained might be different. What would we name packages if we extended this idea to shared frameworks in the future?

I know its a potential rabbit hole and out of scope for now....but I think worth considering as something we might do, and we should consider naming in the context of this idea.

Reply:
I think this is a very interesting scenario, but out of scope for this design.

We would never duplicate the shared runtimes in the global folder. If non-admin runtime acquisition were done with "download-only packages" / "packs", it would need some host support to load from the package cache or something. 

We're not going to do this for 3.0, but we should be sure we aren't doing anything that closes the door to that. Do you think that there's anything here that does?

Reply:
Current status as of Preview 5 - AspNetCore updated to the new name schema, but nothing else has yet.

- runtime.win-x64.Microsoft.NETCore.App
- runtime.win-x64.Microsoft.NETCore.DotNetHostResolver
- runtime.win-x64.Microsoft.NETCore.DotNetHostPolicy
- runtime.win-x64.Microsoft.WindowsDesktop.App
- Microsoft.AspNetCore.App.Runtime.win-x64

cc @dagood